lib/systems/include/MainChargeSystem.h

```diff
@@ -0,0 +1,40 @@
+#ifndef __MAINCHARGE_H__
+#define __MAINCHARGE_H__
+
+
+#include "EthernetInterface.h"
+
+
+/**
+ * MainChargeSystem has its own PID object for adjusting error
+ * It needs to have a function that is a one loop of charging where CCU gets info about the cells, calculates correct current,
+ * and sends out current set and whether to enable charging or not.
+*/
+class MainChargeSystem {
+  public:
+    /**
+     * @param target_volt The wanted final voltage for an individual cell
+     * @param max_charge_current The maximum charge current that should be given to ACU
+     * @param max_cell_temp The maximum allowable cell temperature while charging
+     * @param pid_kp The Kp parameter for the PID that controls charging current with respect to how close the max cell is to the target voltage
+     * @param pid_ki The Ki parameter for the PID that controls charging current with respect to how close the max cell is to the target voltage
+     * @param pid_kd The Kd parameter for the PID that controls charging current with respect to how close the max cell is to the target voltage
```

Comment:
You should be constantly-updating your documentation comments to match the behavior of the code